pkg/apis/networking/v1alpha1/ingress_lifecycle.go

```diff
@@ -63,11 +63,22 @@ func (is *IngressStatus) MarkResourceNotOwned(kind, name string) {
 
 // MarkLoadBalancerReady marks the Ingress with IngressConditionLoadBalancerReady,
 // and also populate the address of the load balancer.
-func (is *IngressStatus) MarkLoadBalancerReady(lbs []LoadBalancerIngressStatus) {
+func (is *IngressStatus) MarkLoadBalancerReady(lbs []LoadBalancerIngressStatus, publicLbs []LoadBalancerIngressStatus, privateLbs []LoadBalancerIngressStatus) {
 	is.LoadBalancer = &LoadBalancerStatus{
 		Ingress: []LoadBalancerIngressStatus{},
 	}
 	is.LoadBalancer.Ingress = append(is.LoadBalancer.Ingress, lbs...)
+
+	is.PublicLoadBalancer = &LoadBalancerStatus{
```

Comment:
Why not just combine the statements?
And fix the old one while you're at it.

`is.LoadBalancer = &LBS{Ingress:lbs}` ?